### PR TITLE
Say no to legacy react wrappers

### PR DESCRIPTION
**Pull request by sanyavertolet**
### What's done:
 - [x] Moved all `fc`s to `FC`s
 - [ ] Partly moved all `RComponent`s to `Component`s

### CODE REVIEW

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/externals/modal/Modal.kt [hunk 1]
```diff
@@ -25,6 +25,7 @@ private val defaultModalStyle = Styles(
 /**
  * @param handler builder for modal component
  */
+@Deprecated("Use non-legacy code")
```
Comment (petertrr) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/externals/modal/Modal.kt:28]: I believe in this PR we'll need to completely remove dependency on `-legacy` artifact, if it's possible :)

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/externals/chart/PieChartBuilder.kt [hunk 2]
```diff
@@ -14,15 +14,15 @@ import kotlin.random.Random
...
     data: Array<DataPieChart>,
-    handler: RHandler<PieChartProps> = {},
-) = child(PieChart::class) {
-    attrs.data = data
-    attrs.animate = false
-    attrs.segmentsShift = 0
-    attrs.viewBoxSize = intArrayOf(100, 100)
-    attrs.radius = 50
+    handler: ChildrenBuilder.(props: PieChartProps) -> Unit = {},
+) = PieChart::class.react.create {
```
Comment (petertrr) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/externals/chart/PieChartBuilder.kt:20]: I don't think it should be `create`, but rather extension `ElemenType.invoke` from `ChildrenBuilder` should be used

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/CreateOrganizationView.kt [hunk 3]
```diff
@@ -127,30 +132,43 @@ class CreateOrganizationView : AbstractView<Props, OrganizationSaveViewState>(tr
...
+                                    height = 260.0
+                                    width = 260.0
                                 }
-                                form(classes = "needs-validation") {
+                                form {
+                                    className = ClassName("needs-validation")
                                     div {
                                         inputTextFormRequired(InputTypes.ORGANIZATION_NAME, state.isValidOrganizationName!!, "", "Organization name", true) {
-                                            changeFields(InputTypes.ORGANIZATION_NAME, it)
+                                            changeFields(InputTypes.ORGANIZATION_NAME, it as Event)
```
Comment (petertrr) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/CreateOrganizationView.kt:164]: What is the inferred type of `it` here?

**Review by petertrr — CHANGES_REQUESTED** [review]
Since the proposed change is too large, I suggest to split it into two parts: right now we can merge `fc -> FC` migration and add `fc` into Detekt's forbidden list (`ForbiddenMethodCall`, `ForbiddenImport` and similar rules), and then we can migrate class components

**Comment by sanyavertolet** [comment]
As far as this PR was split into two different (#907 and #913), it should be considered as done.

**Pull request closed without merging**